win: port poolsets tests

wlemkows opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/1493)
<!-- Reviewable:end -->

krzycz commented:
Reviewed 17 of 17 files at r1.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[src/common/file.c, line 446 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ-VZK8lLCmrfXh2ko:-KYJ-VZK7bkUJ4UsXKaQ:b-qquzrw) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/common/file.c#L446)):*
> ```C
> 	} else {
> #ifdef _WIN32
> 		_chmod(path, _S_IREAD | _S_IWRITE);
> ```

a comment?

---

*[src/test/pmempool_create/TEST6w.PS1, line 64 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ0IhA0OD7y5IQ5k46:-KYJ0IhA7SmwzQuIyve1:bq2h0jb) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L64)):*
> ```
> $REPS="$REPL1 $REPL2 $REPL3"
> 
> remove_files $POOLSET
> ```

I'm not against new function - just our of curiosity - isn't `rm -Force -ea si ...` enough, like we already do in many other tests?

---

*[src/test/pmempool_create/TEST6w.PS1, line 65 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ-re666HdSFCAg8UZ:-KYJ-re6BsFBUqZmKfeG:b-hyb3gj) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L65)):*
> ```
> 
> remove_files $POOLSET
> create_poolset $POOLSET 32M:$DIR\pool.part1:z 32M:$DIR\pool.part2:z 32M:$DIR\pool.part3:z
> ```

Why not " 32M:${POOL1}:z ...", etc. ? (same in other places/scripts)

---

*[src/test/pmempool_create/TEST6w.PS1, line 69 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ01dcELwlFd_iHcNA:-KYJ01dcCYulbRcdrLWq:b-8jmhvl) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L69)):*
> ```
> 
> remove_files $POOLREP
> create_poolset $POOLREP 332M:$DIR\pool.part1:x 32M:$DIR\pool.part2:x 32M:$DIR\pool.part3:x `
> ```

332M?

---

*[src/test/pmempool_info/TEST17w.PS1, line 53 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ3Hio7nFkIQlFAE3Q:-KYJ3Hio8L31CgygwANg:b-gsqucr) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_info/TEST17w.PS1#L53)):*
> ```
> 
> $LOG="out$Env:UNITTEST_NUM.log"
> rm $LOG -Force -ea si
> ```

remove_files? ;-)

---

*[src/test/pmempool_info/TEST17w.PS1, line 63 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ3LUCEdLZG4JtGbqg:-KYJ3LUC4btc1q96ZWIz:b-7ropln) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_info/TEST17w.PS1#L63)):*
> ```
> $REPS="$REPL1 $REPL2"
> 
> rm $POOLSET -Force -ea si
> ```

ditto

---

*[src/test/pmempool_rm/out1w.log.match, line 2 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ3VIS3rAMH57z1NVj:-KYJ3VIS-nLPBLITOzxG:bh4h2dk) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/out1w.log.match#L2)):*
> ```
> $(*)
> remove write-protected file '$(nW)\pool.log' ? [Y/n] n
> ```

Do we need a windows-specific match file? Just remove backslahes - i.e. '$(nW)pool.blk' and this should work on both os'es.

---

*[src/test/pmempool_rm/out2w.log.match, line 2 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4LxN7hCzAfFIVfQY:-KYJ4LxN11cDwhTr51CK:b3kre) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/out2w.log.match#L2)):*
> ```
> $(nW)pmempool$(nW) : cannot remove '$(*)pool.obj'
> $(*)
> ```

?

---

*[src/test/pmempool_rm/TEST2w.PS1, line 70 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4FIMBp93D9_-cHoG:-KYJ4FIM0mgnmdGX3n1I:b-ssab7k) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/TEST2w.PS1#L70)):*
> ```
> 	$DIR\rep.part2
> 
> # XXX: it is workaround for prevent line breaks from out stderr
> ```

Does it work? It looks pretty strange.

---

*[src/test/unittest/unittest.ps1, line 192 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4vas16hwM9CbQJaA:-KYJ4vas19YhVTRiPoki:b9hj42f) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L192)):*
> ```PowerShell
>     [int64]$size = (convert_to_bytes $args[0])
>     # it causes CreateFile with CREATE_ALWAYS flag
>     $mode = "-f"
> ```

Is that required in any test?

---

*[src/test/unittest/unittest.ps1, line 374 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ599hDErSR_kk7bH5:-KYJ599hByloBEnK-0PT:b-odsdgw) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L374)):*
> ```PowerShell
>     # that we can't do, investigating how to address API hooking...
> 
> 	$iterator = 0
> ```

Any comment on what is it for, and how it works?

---

*[src/test/unittest/unittest.ps1, line 692 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ5MzD3W0ROpRFLUGJ:-KYJ5MzD763IF6Vwk2J6:bc6a77) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L692)):*
> ```PowerShell
> #
> function remove_files {
>     for ($i=0;$i -lt $args.count;$i++) {
> ```

Naive question - wouldn't this single-liner work?
`rm -Force -ea si $args`

---

*[src/test/unittest/unittest.ps1, line 750 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ6L4W2ej7YZecIfW0:-KYJ6L4WDOptkI46bbOC:b-cjvm1z) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L750)):*
> ```PowerShell
> #
> # set_file_mode - set access mode to files
> #
> ```

syntax/usage ?

---

*[src/test/unittest/unittest.ps1, line 830 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ64ji1Ar6ijFhXzU5:-KYJ64jiBP4erXZJW7Vd:bb6mt9b) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L830)):*
> ```PowerShell
> #
> function check_signatures {
>     for ($i=1;$i -lt $args.count;$i+=1) {
> ```

:+1:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[src/test/pmempool_create/TEST6w.PS1, line 64 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ0IhA0OD7y5IQ5k46:-KYJ9odHo77S2ozYpsil:b-gk8228) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L64)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

I'm not against new function - just our of curiosity - isn't `rm -Force -ea si ...` enough, like we already do in many other tests?
</blockquote></details>

Can you give me an example of test where we use rm on more than one file?
I tried simply remove variable like $POOLS="$POOL1 $POOL2 $POOL3", but it is treat like one path, not 3 as we expect and that causes errors, so I wrote remove_files func.

---

*[src/test/pmempool_create/TEST6w.PS1, line 65 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ-re666HdSFCAg8UZ:-KYJBt8bcRnTswbl7wbb:b40gk3h) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L65)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Why not " 32M:${POOL1}:z ...", etc. ? (same in other places/scripts)
</blockquote></details>

It does not work in that way, I saw that in other tests we do the same.

---

*[src/test/pmempool_rm/TEST2w.PS1, line 70 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4FIMBp93D9_-cHoG:-KYJCDLz6M7hVaDdxawx:bha7y1x) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/TEST2w.PS1#L70)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Does it work? It looks pretty strange.
</blockquote></details>

Yes, it changes console buffer width and thanks to this lines of standard error output aren't break and match file match.

---

*[src/test/unittest/unittest.ps1, line 192 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4vas16hwM9CbQJaA:-KYJDp8S8Mce8MiO9WKZ:b-bogkgw) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L192)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Is that required in any test?
</blockquote></details>

I this so, on Linux we uses truncate in that case, and it always create file, even if exist.

---

*[src/test/unittest/unittest.ps1, line 692 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ5MzD3W0ROpRFLUGJ:-KYJFwBz7qL6J6_DvFEt:bw23626) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L692)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Naive question - wouldn't this single-liner work?
`rm -Force -ea si $args`
</blockquote></details>

Look comment above.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[src/test/pmempool_create/TEST6w.PS1, line 64 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ0IhA0OD7y5IQ5k46:-KYJuhSI-0vK8RRKXz8q:b750re1) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L64)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

Can you give me an example of test where we use rm on more than one file?
I tried simply remove variable like $POOLS="$POOL1 $POOL2 $POOL3", but it is treat like one path, not 3 as we expect and that causes errors, so I wrote remove_files func.
</blockquote></details>

Right, you can use wildcards to i..e delete all the "txt" files, but it's not allowed to specify more than one name.

---

*[src/test/pmempool_create/TEST6w.PS1, line 65 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ-re666HdSFCAg8UZ:-KYJuqS1CYdShej1H6WJ:bcixmyg) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L65)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

It does not work in that way, I saw that in other tests we do the same.
</blockquote></details>

Yes, it wouldn't work w/o braces, as the var name is followed by ":". But with braces, I think it should work fine (tested).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: 3 of 19 files reviewed at latest revision, 14 unresolved discussions.

---

*[src/common/file.c, line 446 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ-VZK8lLCmrfXh2ko:-KYTI5SnCKq6040FvKW7:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/common/file.c#L446)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

a comment?
</blockquote></details>

Done.

---

*[src/test/pmempool_create/TEST6w.PS1, line 65 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ-re666HdSFCAg8UZ:-KYOGSJlD5I3C9lFYdnm:b-1nqtx1) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L65)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Yes, it wouldn't work w/o braces, as the var name is followed by ":". But with braces, I think it should work fine (tested).
</blockquote></details>

Thanks, it works. Done.

---

*[src/test/pmempool_create/TEST6w.PS1, line 69 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ01dcELwlFd_iHcNA:-KYO9uv_CwewlJ80EyjL:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_create/TEST6w.PS1#L69)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

332M?
</blockquote></details>

Done.

---

*[src/test/pmempool_info/TEST17w.PS1, line 53 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ3Hio7nFkIQlFAE3Q:-KYONuTMBrSVz9m3lCLo:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_info/TEST17w.PS1#L53)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

remove_files? ;-)
</blockquote></details>

Done.

---

*[src/test/pmempool_info/TEST17w.PS1, line 63 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ3LUCEdLZG4JtGbqg:-KYONvkp-YyWvVSiIWT0:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_info/TEST17w.PS1#L63)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[src/test/pmempool_rm/out2w.log.match, line 2 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4LxN7hCzAfFIVfQY:-KYObmKTB9xa4NYWI9KE:b-iw1gj1) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/out2w.log.match#L2)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

?
</blockquote></details>

On windows 2>&1 looks different than on Linux. I changed regex in test for simplicity this match.

---

*[src/test/unittest/unittest.ps1, line 374 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ599hDErSR_kk7bH5:-KYSssm5EKQljD-3aeSO:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L374)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Any comment on what is it for, and how it works?
</blockquote></details>

Done.

---

*[src/test/unittest/unittest.ps1, line 750 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ6L4W2ej7YZecIfW0:-KYTMfAP5jxDy-Tqgvqj:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L750)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

syntax/usage ?
</blockquote></details>

Done.

---

*[src/test/unittest/unittest.ps1, line 830 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ64ji1Ar6ijFhXzU5:-KYP642-1xinwkoaCtfO:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L830)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

:+1:
</blockquote></details>

Done.

---

*[src/test/pmempool_rm/out1w.log.match, line 2 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ3VIS3rAMH57z1NVj:-KYOUQYJ8AGy-9ysrnFQ:b-79xyid) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/out1w.log.match#L2)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Do we need a windows-specific match file? Just remove backslahes - i.e. '$(nW)pool.blk' and this should work on both os'es.
</blockquote></details>

Fixed. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Reviewed 2 of 19 files at r1, 21 of 21 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[src/test/pmempool_info/TEST17.PS1, line 33 at r2](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYVC5SsBEpWhe9rooXz:-KYVC5SsBEpWhe9rooY-:b32v8v) ([raw file](https://github.com/pmem/nvml/blob/97ce0cfb3ced3a1f0cb6b7830260d242b2b1951b/src/test/pmempool_info/TEST17.PS1#L33)):*
> ```
> #
> #
> # pmempool_info\TEST17w.PS1 -- test for info command
> ```

-w

---

*[src/test/pmempool_rm/TEST1.PS1, line 80 at r2](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYYQGxP0GbYK6smZeHd:-KYYQGxP0GbYK6smZeHe:b-q1hbtn) ([raw file](https://github.com/pmem/nvml/blob/97ce0cfb3ced3a1f0cb6b7830260d242b2b1951b/src/test/pmempool_rm/TEST1.PS1#L80)):*
> ```
> 	$DIR\rep.part2
> 
> expect_normal_exit -p 'n`nn`nn`ny`ny`ny`ny`n' $PMEMPOOL rm -sv `
> ```

See ex_libpmemobj tests to see how to redirect string to stdin.

---

*[src/test/pmempool_rm/TEST2w.PS1, line 70 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4FIMBp93D9_-cHoG:-KYVCNIjnzPAN2NGeike:brnqusf) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/TEST2w.PS1#L70)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

Yes, it changes console buffer width and thanks to this lines of standard error output aren't break and match file match.
</blockquote></details>

`... | out-file -width 1024 $file`
or better:
`... | %{$_.Line} > $file`

---

*[src/test/unittest/unittest.ps1, line 374 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ599hDErSR_kk7bH5:-KYYPUOQp4BQSX8_e9I8:b9iposp) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L374)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Do we need this? See ex_libpmemobj.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[src/test/pmempool_rm/TEST1.PS1, line 80 at r2](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYYQGxP0GbYK6smZeHd:-KYYX6OV6SP_ScYSWo_i:b-1btzqz) ([raw file](https://github.com/pmem/nvml/blob/97ce0cfb3ced3a1f0cb6b7830260d242b2b1951b/src/test/pmempool_rm/TEST1.PS1#L80)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

See ex_libpmemobj tests to see how to redirect string to stdin.
</blockquote></details>

So I don't have to use expect_normal_exit in TEST file to execute $PMEMPOOL file?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: 17 of 21 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/test/pmempool_info/TEST17.PS1, line 33 at r2](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYVC5SsBEpWhe9rooXz:-KYYg5RJKb3CkaPUnQ5y:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/97ce0cfb3ced3a1f0cb6b7830260d242b2b1951b/src/test/pmempool_info/TEST17.PS1#L33)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

-w
</blockquote></details>

Done.

---

*[src/test/pmempool_rm/TEST1.PS1, line 80 at r2](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYYQGxP0GbYK6smZeHd:-KYYi6zYNhRlXCJ-wU_-:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/97ce0cfb3ced3a1f0cb6b7830260d242b2b1951b/src/test/pmempool_rm/TEST1.PS1#L80)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

So I don't have to use expect_normal_exit in TEST file to execute $PMEMPOOL file?
</blockquote></details>

Done.

---

*[src/test/pmempool_rm/TEST2w.PS1, line 70 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ4FIMBp93D9_-cHoG:-KYYfh93LXAFxImmqpZy:bsvjh7u) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/pmempool_rm/TEST2w.PS1#L70)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

`... | out-file -width 1024 $file`
or better:
`... | %{$_.Line} > $file`
</blockquote></details>

Done. out-file -width 1024 $file - it works for 2>&1

---

*[src/test/unittest/unittest.ps1, line 374 at r1](https://reviewable.io:443/reviews/pmem/nvml/1493#-KYJ599hDErSR_kk7bH5:-KYYfgEGmce7fNGyOHX_:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/9ed52ad85928b3515ac0f3c029c5ff5bd6a4da07/src/test/unittest/unittest.ps1#L374)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Do we need this? See ex_libpmemobj.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493)*
<!-- Sent from Reviewable.io -->

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 21 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493#-:-KYZQmhl4Fxgy-tJlt-D:bnfp4nl)*
<!-- Sent from Reviewable.io -->

plebioda commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 5 of 19 files at r1, 16 of 21 files at r2, 4 of 4 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1493#-:-KZ3AMGkPj9PREN1Mfp0:bnfp4nl)*
<!-- Sent from Reviewable.io -->
